### ISSUE TITLE
clickhouse-test continues working on SIGINT

### ISSUE THREAD

**Issue by myrrc** [ClickHouse/ClickHouse#10982]
(you don't have to strictly follow this form)

**Describe the bug**
./clickhouse test does not interrupt on SIGINT, continues looping.

**How to reproduce**
* Which ClickHouse server version to use -- last master

**Expected behavior**
clickhouse-test terminating

**Error message and/or stacktrace**
```
KeyboardInterrupt
    self.run()
    racquire()
  File "/usr/lib/python2.7/multiprocessing/process.py", line 114, in run
KeyboardInterrupt
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python2.7/multiprocessing/pool.py", line 102, in worker
    self.run()
    task = get()
  File "/usr/lib/python2.7/multiprocessing/process.py", line 114, in run
  File "/usr/lib/python2.7/multiprocessing/process.py", line 267, in _bootstrap
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python2.7/multiprocessing/pool.py", line 102, in worker
    self.run()
  File "/usr/lib/python2.7/multiprocessing/process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python2.7/multiprocessing/queues.py", line 374, in get
  File "/usr/lib/python2.7/multiprocessing/pool.py", line 102, in worker
    task = get()
  File "/usr/lib/python2.7/multiprocessing/queues.py", line 374, in get
    racquire()
KeyboardInterrupt
    task = get()
  File "/usr/lib/python2.7/multiprocessing/queues.py", line 374, in get
    racquire()
KeyboardInterrupt
    racquire()
KeyboardInterrupt

```

**Additional context**

**Comment by akuzm** [comment]
Oh, it's just a giant mess. I use Ctrl+Z + kill %, works kinda better. Tried to fix it once, but broke something else:

573983d4079b280149699c5fe2b522f4ae4c9fb4

**Comment by myrrc** [comment]
> Oh, it's just a giant mess, there are several python bugs and our bugs conspiring to make proper termination impossible. I use Ctrl+Z + kill %, works kinda better. Tried to fix it once, but broke something else: [573983d](https://github.com/ClickHouse/ClickHouse/commit/573983d4079b280149699c5fe2b522f4ae4c9fb4) Also this fix was for python 3, and now this script uses python 2 for some reason.

@akuzm 
If someone hels me with the first PR https://github.com/ClickHouse/ClickHouse/pull/10950, I can help finish porting the tester to python3.

IMHO The ticket is not resolved, so just waiting till some fixes it :).
